### ISSUE TITLE
GUI libraries

### ISSUE THREAD

**Issue by Dekken** [darlinghq/darling#391]
Hi there, 

is there a way to disable all UI stuff? 

The default build looks for GL and not EGL, but requires EGL if it finds GL

I just want the shell 

Thanks

**Comment by bugaevc** [comment]
Yes. At the moment, we don't have a better way to disable parts of the build than manually editing `src/CMakeLists.txt`. You'd want to comment out (with a `#`) the `src/OpenGL` line and everything starting with `src/external/cocotron`.

While you're at it, you may also want to disable `xcbuild`, `JavaScriptCore` (JSC is huge, so takes a lot of time and disk space to build) and maybe some versions of Python, Ruby and Perl you don't need.

**Comment by bugaevc** [comment]
`src/native` (`wrap_elf` stuff) as well, I guess.

I think we should recommend distro packagers to split Darling into subpackages, e.g. `darling-lkm`, `darling-core` (libSystem & stuff), `darling-unix` (misc Unix commands, manuals, etc), `darling-cocoa`, `darling-backend-gtk3` and so on, and then make an umbrella `darling` package that would depend on a selection (all?) of them.

**Comment by Dekken** [comment]
could it be done from here? i.e. from a source perpsetive

**Comment by bugaevc** [comment]
Yes, we could provide configure-time options for disabling parts of the build (there used to be `-DFRAMEWORK_APPKIT` and `-DFRAMEWORK_COREAUDIO` some time ago...)

We'd gladly accept a patch ;)
